To: vim_dev@googlegroups.com
Subject: Patch 7.3.436
Fcc: outbox
From: Bram Moolenaar <Bram@moolenaar.net>
Mime-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit
------------

Patch 7.3.436
Problem:    Compiler warnings for types on Windows.
Solution:   Add type casts. (Mike Williams)
Files:      src/eval.c


*** ../vim-7.3.435/src/eval.c	2012-02-06 00:13:16.000000000 +0100
--- src/eval.c	2012-02-11 20:38:16.000000000 +0100
***************
*** 14462,14468 ****
  			    --prevlen;
  		}
  		if (prevlen == 0)
! 		    s = vim_strnsave(start, len);
  		else
  		{
  		    /* Change "prev" buffer to be the right size.  This way
--- 14462,14468 ----
  			    --prevlen;
  		}
  		if (prevlen == 0)
! 		    s = vim_strnsave(start, (int)len);
  		else
  		{
  		    /* Change "prev" buffer to be the right size.  This way
***************
*** 14529,14535 ****
  
  			if (dest < buf)
  			{
! 			    adjust_prevlen = buf - dest; /* must be 1 or 2 */
  			    dest = buf;
  			}
  			if (readlen > p - buf + 1)
--- 14529,14535 ----
  
  			if (dest < buf)
  			{
! 			    adjust_prevlen = (int)(buf - dest); /* must be 1 or 2 */
  			    dest = buf;
  			}
  			if (readlen > p - buf + 1)
***************
*** 14558,14568 ****
  		 * small, to avoid repeatedly 'allocing' large and
  		 * 'reallocing' small. */
  		if (prevsize == 0)
! 		    prevsize = p - start;
  		else
  		{
  		    long grow50pc = (prevsize * 3) / 2;
! 		    long growmin  = (p - start) * 2 + prevlen;
  		    prevsize = grow50pc > growmin ? grow50pc : growmin;
  		}
  		if ((newprev = vim_realloc(prev, prevsize)) == NULL)
--- 14558,14568 ----
  		 * small, to avoid repeatedly 'allocing' large and
  		 * 'reallocing' small. */
  		if (prevsize == 0)
! 		    prevsize = (long)(p - start);
  		else
  		{
  		    long grow50pc = (prevsize * 3) / 2;
! 		    long growmin  = (long)((p - start) * 2 + prevlen);
  		    prevsize = grow50pc > growmin ? grow50pc : growmin;
  		}
  		if ((newprev = vim_realloc(prev, prevsize)) == NULL)
***************
*** 14575,14581 ****
  	    }
  	    /* Add the line part to end of "prev". */
  	    mch_memmove(prev + prevlen, start, p - start);
! 	    prevlen += p - start;
  	}
      } /* while */
  
--- 14575,14581 ----
  	    }
  	    /* Add the line part to end of "prev". */
  	    mch_memmove(prev + prevlen, start, p - start);
! 	    prevlen += (long)(p - start);
  	}
      } /* while */
  
*** ../vim-7.3.435/src/version.c	2012-02-11 20:40:49.000000000 +0100
--- src/version.c	2012-02-11 20:43:18.000000000 +0100
***************
*** 716,717 ****
--- 716,719 ----
  {   /* Add new patch number below this line */
+ /**/
+     436,
  /**/

-- 
hundred-and-one symptoms of being an internet addict:
39. You move into a new house and decide to Netscape before you landscape.

 /// Bram Moolenaar -- Bram@Moolenaar.net -- http://www.Moolenaar.net   \\\
///        sponsor Vim, vote for features -- http://www.Vim.org/sponsor/ \\\
\\\  an exciting new programming language -- http://www.Zimbu.org        ///
 \\\            help me help AIDS victims -- http://ICCF-Holland.org    ///
